streams/src/test/java/org/apache/kafka/streams/kstream/internals/WindowStoreAdaptTest.java

```diff
@@ -0,0 +1,150 @@
+
+    @Test
+    public void testWithKVStore() {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        builder
+            .stream("input", Consumed.with(Serdes.String(), Serdes.String()))
+            .groupByKey()
+            .windowedBy(TimeWindows.of(ofMillis(100)))
+            .aggregate(
+                () -> "init",
+                (k1, v1, prev) -> prev + " (" + k1 + "," + v1 + ")",
+                Materialized.<String, String, WindowStore<Bytes, byte[]>>with(Serdes.String(), Serdes.String())
+            )
+            .filter(
+                (windowedKey, agg) -> agg.split(" ").length % 2 == 0,
+                MaterializedKV.<Windowed<String>, String>as(Stores.inMemoryKeyValueStore("store"))
+                    .withKeySerde(WindowedSerdes.timeWindowedSerdeFrom(String.class))
+                    .withValueSerde(Serdes.String())
+            )
+            .toStream()
+            .foreach((k, v) -> System.out.println("filtered K: " + k + ", V: " + v));
+        final Topology topology = builder.build();
+
+
+        final ConsumerRecordFactory<String, String> recordFactory =
+            new ConsumerRecordFactory<>(STRING_SERIALIZER, STRING_SERIALIZER);
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(topology, config)) {
+            driver.pipeInput(recordFactory.create("input", "k1", "v1", 0L));
+            driver.pipeInput(recordFactory.create("input", "k1", "v2", 1L));
+            driver.pipeInput(recordFactory.create("input", "k2", "v1", 2L));
+        }
+    }
+
+    @Test
+    public void testWithWindowedStore() {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        builder
+            .stream("input", Consumed.with(Serdes.String(), Serdes.String()))
+            .groupByKey()
+            .windowedBy(TimeWindows.of(ofMillis(100)))
+            .aggregate(
+                () -> "init",
+                (k1, v1, prev) -> prev + " (" + k1 + "," + v1 + ")",
+                Materialized.<String, String, WindowStore<Bytes, byte[]>>with(Serdes.String(), Serdes.String())
+            )
+            .filter(
+                (windowedKey, agg) -> agg.split(" ").length % 2 == 0,
+                MaterializedKV.<Windowed<String>, String>as(Stores.inMemoryWindowStore("store", Duration.ofDays(1), Duration.ofMillis(100), false))
+                    .withKeySerde(WindowedSerdes.timeWindowedSerdeFrom(String.class))
+                    .withValueSerde(Serdes.String())
+            )
+            .toStream()
+            .foreach((k, v) -> System.out.println("filtered K: " + k + ", V: " + v));
+        final Topology topology = builder.build();
+
+
+        final ConsumerRecordFactory<String, String> recordFactory =
+            new ConsumerRecordFactory<>(STRING_SERIALIZER, STRING_SERIALIZER);
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(topology, config)) {
+            driver.pipeInput(recordFactory.create("input", "k1", "v1", 0L));
```

Comment:
If we call `getWindowStore`, it will give us a null because of type mismatch (expected window store, actually timestamped key-value store)

Reply:
This is a good point. When we fetch the store for IQ and for TTD, we would unwrap it and just present it as a WindowStore.